pkgs/tools/security/jadx-bin/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, makeWrapper, jre, unzip, makeDesktopItem }:
+
+stdenv.mkDerivation rec {
+  version = "0.9.0";
+  name = "jadx-bin-${version}";
+
+  desktopItem = makeDesktopItem {
+    name = "jadx-gui";
+    exec = "jadx-gui %F";
+    comment = "Java Decompiler";
+    desktopName = "jadx GUI";
+    genericName = "Java and Android Decompiler";
+    mimeType = "application/x-java-archive;application/x-java";
+    categories = "Development;Debugger;";
+  };
+
+  meta = with stdenv.lib; {
+    description = "Dex to Java decompiler";
+    homepage    = "https://github.com/skylot/jadx";
+    license     = licenses.asl20;
+    platforms   = platforms.unix;
+    maintainers = [ maintainers.artemist ];
+  };
+
+  src = fetchurl {
+    url    = "https://github.com/skylot/jadx/archive/v${version}/jadx-${version}.zip";
```

Comment:
This url doesn't work for me, it downloads the sources archive instead. I had to change it to `https://github.com/skylot/jadx/releases/download/v${version}/jadx-${version}.zip` which downloads the binary archive and matches the hash.

I've noticed with nix that if you change the url but not the hash it will continue to use the previous source associated with that hash (issue #969). This would explain why it can build successfully for you with that url.